src/components/LinkInput/LinkInput.tsx

```diff
@@ -1,50 +1,30 @@
 import { FormEvent,ChangeEvent, useState, useEffect } from 'react';
 import { getShortedLink } from '../../helpers/getShortedLink';
-import { LinkResponseData } from '../../types/types';
+import { LinkInputProps, LinkResponseData } from '../../types/types';
+import { validateLink } from '../../helpers/validateLink';
 
-type FormProps = {
-  setShortenedLink: (value:LinkResponseData) => void;
-};
-
-export const LinkInput = ({setShortenedLink}:FormProps) => {
+export const LinkInput = ({ shortenedLink, setShortenedLink}:LinkInputProps) => {
   const [inputValue, setInputValue] = useState('');
   const [errorMsg, setErrorMsg] = useState('');
 
   useEffect(() => {
-    const storedLinks:LinkResponseData = JSON.parse(localStorage.getItem('links')!);
+    const storedLinks:LinkResponseData[] = JSON.parse(localStorage.getItem('links')!) || [];
```

Comment:
Si bien, esto es técnicamente viable, da un poco de confusión al ver que siempre habrá algo en "links" cuando tal vez no siempre va a ser así. Para dar más claridad, se podría hacer algo como:

```suggestion
    const storedLinks:LinkResponseData[] | null = localStorage.getItem('links') ? JSON.parse(localStorage.getItem('links')) : null; // Aquí retorno un null debido a la condicional que tienes en la linea inferior
```

Siento que es preferible ser un poco más específicos